src/chainstate/burn/db/sortdb.rs

```diff
@@ -3230,9 +3240,18 @@ impl<'a> SortitionHandleTx<'a> {
                     if reward_set.len() > 0 {
                         // if we have a reward set, then we must also have produced a recipient
                         //   info for this block
-                        let (addr, ix) = recipient_info.unwrap().recipient.clone();
-                        assert_eq!(&reward_set.remove(ix as usize), &addr,
-                            "BUG: Attempted to remove used address from reward set, but failed to do so safely");
+                        let mut recipients_to_remove: Vec<_> = recipient_info
+                            .unwrap()
+                            .recipients
+                            .iter()
+                            .map(|(addr, ix)| (addr.clone(), *ix))
+                            .collect();
+                        recipients_to_remove.sort_unstable_by(|(_, a), (_, b)| b.cmp(a));
```

Comment:
Sorry -- I think you've answered this question before, but I've forgotten the answer -- why is this an unstable sort?  Is this the point in the code that removes addresses from the reward set, and thus must be deterministic?

Reply:
It uses an unstable sort because it can: the indexes it is sorting are unique anyways, so it doesn't matter if the sort preserves the original ordering, and unstable sorts have faster implementations. Though it hardly matters here, the input size is really just 2: this step is sorting the indexes that are getting removed, so that it can remove them without invalidating the next index.

Reply:
Thanks for refreshing my memory!